src/test/java/com/arpnetworking/tsdcore/sinks/KairosDbSinkTest.java

```diff
@@ -143,7 +145,7 @@ public void testPost() throws InterruptedException, IOException {
                 Mockito.any());
         Mockito.verify(_mockMetrics, Mockito.times(1)).startTimer("sinks/http_post/kairosdb_sink_test/request_latency");
         Mockito.verify(_mockMetrics, Mockito.times(1)).stopTimer("sinks/http_post/kairosdb_sink_test/request_latency");
-        Mockito.verify(_mockMetrics, Mockito.times(1)).close();
+        Mockito.verify(_mockMetrics, Mockito.times(2)).close();
     }
```

Comment:
Looks like you're missing a test case for the feature you're trying to add.  Can you please add a test showing the retry on a failed POST?